### PR TITLE
Feat/due dates

### PR DESCRIPTION
**Pull request by EdotJ**
Added a datepicker to pick out a date when the todo is due. 
Added an indicator for todos past their deadline. Todos pending are yellow, todos past - red, todos done - green.

### CODE REVIEW

**Comment by EdotJ** [comment]
[Add indicators for TODOs past deadlines](https://trello.com/c/kTQ8kARz/19-add-indicators-for-todos-past-deadlines)

File: ToNote/Models/ToDo.cs [hunk 1]
```diff
@@ -35,5 +43,31 @@ public int Index
...
+                    RaisePropertyChanged(nameof(IsDatePast));
+                }
+                
+            }
+        }
+
+        public bool IsDatePast
+        {
+            get
+            {
```
Comment (Kodnot) [inline ToNote/Models/ToDo.cs]: To expression-bodied property, ```public bool IsDatePast => DateTime.Compare(_SelectedDate, DateTime.Today) > 0 ? true : false;```

Comment (EdotJ) [reply ToNote/Models/ToDo.cs]: Done.

File: ToNote/Models/ToDo.cs [hunk 2]
```diff
@@ -15,6 +17,14 @@ public bool IsChecked
                 {
                     _IsChecked = value;
 
+                    if (_IsChecked == true)
```
Comment (Kodnot) [inline ToNote/Models/ToDo.cs]: Don't explicitly check bools for true, just `if(_IsChecked)`

File: ToNote/Models/ToDo.cs [hunk 3]
```diff
@@ -35,5 +45,32 @@ public int Index
...
+                    _SelectedDate = value;
+
+                    RaisePropertyChanged(nameof(SelectedDate));
+                    RaisePropertyChanged(nameof(IsDatePast));
+                }
+                
+            }
+        }
+
+        private DateTime? _LastDone;
```
Comment (Kodnot) [inline ToNote/Models/ToDo.cs]: As you don't raise any events here on property changed, you could just make this `public DateTime? LastDone { get; private set; }`

File: ToNote/Models/ToDo.cs [hunk 4]
```diff
@@ -35,5 +45,32 @@ public int Index
...
         }
+
+        private DateTime? _SelectedDate;
+
+        public DateTime? SelectedDate
+        {
+            get => _SelectedDate;
+            set
+            {
+                if (_SelectedDate != value)
```
Comment (Kodnot) [inline ToNote/Models/ToDo.cs]: You could invert this to reduce nesting

File: ToNote/Models/ToDo.cs [hunk 5]
```diff
@@ -35,5 +45,32 @@ public int Index
...
+        }
+
+        private DateTime? _LastDone;
+
+        public DateTime? LastDone
+        {
+            get => _LastDone;
+        }
+
+        public bool IsDatePast => _SelectedDate != null ? DateTime.Compare((DateTime)_SelectedDate, DateTime.Today) > 0 ? true : false : false;
```
Comment (Kodnot) [inline ToNote/Models/ToDo.cs]: Refactor spaghetti to `public bool IsDatePast => _SelectedDate.HasValue ? (DateTime.Compare(_SelectedDate.Value, DateTime.Today) >= 0) : false;`. Notice that I changed > to >=, I don't think a task that is due today should be marked as overdue.

**Review by Kodnot — APPROVED** [review]
Fix these few comments and you can merge

**Pull request merged**